iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -1108,18 +1143,18 @@ private async Task HandleReceivedCloudToDeviceMessageAsync(MqttApplicationMessag
                     if (Logging.IsEnabled)
                         Logging.Error(this, "Cannot 'reject' or 'abandon' a received message over MQTT. Message will be acknowledged as 'complete' instead.");
                 }
-
-                // Note that MQTT does not support Abandon or Reject, so always Complete by acknowledging the message like this.
-                try
-                {
-                    await receivedEventArgs.AcknowledgeAsync(CancellationToken.None).ConfigureAwait(false);
-                }
-                catch (Exception ex)
+            }
+            else if(!_isDeviceReceiveMessageCallbackSet)
+            {
+                _messageQueue.Enqueue(receivedCloudToDeviceMessage);
+                while(_messageQueue.Count > _messageQueueSize)
                 {
-                    // This likely happened because the connection was lost. The service will re-send this message so the user
-                    // can acknowledge it on the new connection.
```

Comment:
Is this comment being removed because it is inaccurate?

Reply:
yes, the service does not re-send the message